src/databricks/sql/client.py

```diff
@@ -297,6 +300,117 @@ def _check_not_closed(self):
         if not self.open:
             raise Error("Attempting operation on closed cursor")
 
+    def _handle_staging_operation(self, uploads_base_path: str):
+        """Fetch the HTTP request instruction from a staging ingestion command
+        and call the designated handler.
+        
+        Raise an exception if localFile is specified by the server but the localFile
+        is not descended from uploads_base_path.
+        """
+
+        if uploads_base_path is None:
+            raise Error(
+                "You must provide an uploads_base_path when initialising a connection to perform ingestion commands"
+            )
+        
+        row = self.active_result_set.fetchone()
+
+        if getattr(row, "localFile", None):
+            if os.path.commonpath([row.localFile, uploads_base_path]) != uploads_base_path:
```

Comment:
what happens if `uploads_base_path = /Users/user1` and `row.localFile = /Users/user1/../user2` what does this method return?

Reply:
could you please add some tests for that?

Reply:
Good catch.

tl;dr I updated the code in 34a0362a81de1c5ceeeaca681d084fb2e9ed5249 so that it resolves any relative paths _before_ checking for their common_path. I added a test to prove this.

## Before

`/Users/user1` and `/Users/user1/../user2` show a common path of `/Users/user1` which is **wrong**.

## After

`/Users/user1` and `/Users/user1/../user2` show a common path of `/Users` which is correct.

Reply:
@moderakh Are there other cases we should consider?